jsapp/js/components/submissions/tableConstants.ts

```diff
@@ -83,10 +105,31 @@ export const TEXT_FILTER_QUESTION_TYPES = [
   META_QUESTION_TYPES['background-audio'],
 ];
 
+/**
+ * For these question ids the UI will display a text filter in Data Table for
+ * the matching column. We need this, because these are additional submission
+ * properties, so they don't have a question type attached to them.
+ */
 export const TEXT_FILTER_QUESTION_IDS = [
   '__version__',
   ADDITIONAL_SUBMISSION_PROPS._id,
   ADDITIONAL_SUBMISSION_PROPS._uuid,
   ADDITIONAL_SUBMISSION_PROPS._submission_time,
   ADDITIONAL_SUBMISSION_PROPS._submitted_by,
 ];
+
+/**
+ * These are question types that will be filtered by the exact filter value
+ * (i.e. filter value is exactly the response). Any question type not on this
+ * list will be filtered by responses that include the value (i.e. filter value
+ * is part of the response).
+ */
+export const FILTER_EXACT_TYPES: AnyRowTypeName[] = [
+  QuestionTypeName.decimal,
```

Comment:
I don't think I understand this exact value change. I'd expect this to mean that, using decimal as an example, if my submissions were `2.2` and `2.3` then the filter would not show either of these as neither are exactly equal to `2`. But I don't see it behaving like that in the data table:

![image](https://github.com/kobotoolbox/kpi/assets/16762675/d6ebdee1-cb74-4df1-bc72-685aafd58308)

Reply:
Actually it looks like all the mentioned exact types still follow the previous scheme of "1" matching "12", "1.1" etc.

Reply:
I've used this [ALL TYPES FORM.xlsx](https://github.com/kobotoolbox/kpi/files/14904929/ALL.TYPES.FORM.xlsx) for testing.

I've tested it now with every question type and it works as expected. All `FILTER_EXACT_TYPES` are matching exact strings, and all others are partial matching.

Reply:
Could you check if your code goes into proper place (near `FILTER_EXACT_TYPES.includes…`) in `buildFilterQuery` function?

Reply:
Looks good now!